src/main/java/core/basesyntax/strategy/PurchaseHandler.java

```
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.models.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void operate(FruitTransaction fruitTransaction) {
+        String fruitName = fruitTransaction.getFruit();
+        int quantityOfParchedFruits = fruitTransaction.getQuantity();
+        int oldQuantity = Storage.fruits.get(fruitName);
+        Storage.fruits.put(fruitName, oldQuantity - quantityOfParchedFruits);
```

lets throw an exception when there is not enough fruits